main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintGen.scala

```diff
@@ -769,8 +769,9 @@ object ConstraintGen {
       case Expr.InvokeConstructor2(clazz, exps, cvar, evar, loc) =>
         val tpe = Type.getFlixType(clazz)
         val (tpes, effs) = exps.map(visitExp).unzip
+        val baseEffs = BaseEffects.of(clazz, loc)
```

Comment:
Should this not also happen for InvokeMethod, etc?

Reply:
Yes, and more. This is a draft for @AStenbaek to work on.